bundles/mapping/camera-controls-3d/view/CameraControls3d.jsx

```diff
@@ -0,0 +1,141 @@
+    cursor: pointer;
+    z-index: 15000;
+    margin-bottom: 5px;
+    margin-left: 5px;
+    box-shadow: ${iconShadow};
+    -moz-box-shadow: ${iconShadow};
+    -webkit-box-shadow: ${iconShadow};
+    -o-box-shadow: ${iconShadow};
+    background-color: ${props => props.controlIsActive && !props.isMobile ? secondaryColor : darkBgColor};
+    opacity: ${props => props.disabled ? 0.3 : 1.0};
+    pointer-events: ${props => props.disabled ? 'none': 'auto'};
+`;
+
+const MapControl = styled.div`
+    width: 32px;
+    height: 32px;
+    background-repeat: no-repeat;
+    margin: 0;
+    padding-top:3px;
+    text-align: center;
+`;
+const MobileContainer = styled.div`
+    margin-top: 5px;
+    margin-right: 5px;
+`;
+
+const Break = styled.div`
+    flex-basis: 100%;
+    height: 0;
+`;
+
+const getMapModule = () => {
+    return Oskari.getSandbox().getStatefulComponents().mapfull.getMapModule();
+};
+
+const upDownClickHandler = (directionUp) => {
+    const mapmodule = getMapModule();
+    const cam = mapmodule.getCamera();
+    if (directionUp) {
+        cam.location.altitude = cam.location.altitude * ((100 + upDownChangePercent) / 100);
+    } else {
+        cam.location.altitude = cam.location.altitude * ((100 - upDownChangePercent) / 100);
+    }
+    mapmodule.setCamera(cam);
+    Oskari.getSandbox().postRequestByName('MapMoveRequest');
+};
+
+const setCameraToMoveMode = (activeMapMoveMethod) => {
+    if (activeMapMoveMethod === mapMoveMethodMove) {
+        return;
+    }
+    getMapModule().setCameraToMoveMode();
+};
+const setCameraToRotateMode = (activeMapMoveMethod) => {
+    if (activeMapMoveMethod === mapMoveMethodRotate) {
+        return;
+    }
+    getMapModule().setCameraToRotateMode();
+};
+
+const CameraControls3d = ({ mapInMobileMode, getMessage }) => {
+    const [activeMapMoveMethod, setActiveMapMoveMethod] = useState(mapMoveMethodMove);
+
+    const moveMapControl = <MoveMapIcon mapInMobileMode={mapInMobileMode} clickHandler={() => {
+        setActiveMapMoveMethod(mapMoveMethodMove);
+        setCameraToMoveMode();
+    }} title={mapInMobileMode ? '' : getMessage('tooltip.move')} controlIsActive = {activeMapMoveMethod === mapMoveMethodMove}/>;
+
+    const rotateMapControl = <RotateMapIcon mapInMobileMode={mapInMobileMode} clickHandler={() => {
+        setActiveMapMoveMethod(mapMoveMethodRotate);
+        setCameraToRotateMode();
+    }} title={ mapInMobileMode ? '' : getMessage('tooltip.rotate')} controlIsActive = {activeMapMoveMethod === mapMoveMethodRotate}/>;
+
+    const upControl = <UpIcon mapInMobileMode={mapInMobileMode}
+        clickHandler={() => upDownClickHandler(true)} title={mapInMobileMode ? '' : getMessage('tooltip.up')}/>;
+    const downControl = <DownIcon mapInMobileMode={mapInMobileMode}
+        clickHandler={() => upDownClickHandler(false)} title={mapInMobileMode ? '' : getMessage('tooltip.down')}/>;
+
+    let controls;
+
+    if (!mapInMobileMode) {
+        controls = <MapControlsContainer>
```

Comment:
If you return here you can remove the "controls" variable.

Reply:
Done